### PR TITLE
chore: fixes example by not reusing the transaction.

### PR DESCRIPTION
**Pull request by jcchavezs**
Currently, the example was reusing the transaction which ended up in a double closing of body buffer and hence nil pointer exception.

Ping @M4tteoP 

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: examples/http-server/main.go [hunk 1]
```diff
@@ -44,8 +45,8 @@ func setupCoraza() error {
 }
 
 func corazaRequestHandler(h http.Handler) http.Handler {
-	tx := waf.NewTransaction(context.Background())
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		tx := waf.NewTransaction(context.Background())
```
Comment (jcchavezs) [inline examples/http-server/main.go:49]: This is the important change.

File: body_buffer.go [hunk 2]
```diff
@@ -59,11 +75,20 @@ func (br *BodyBuffer) Reader() (io.Reader, error) {
...
+	if br.isClosed {
+		return 0
+	}
+
 	return br.length
 }
 
 // Close will close all readers and delete temporary files
 func (br *BodyBuffer) Close() error {
+	if br.isClosed {
```
Comment (jcchavezs) [inline body_buffer.go:87]: `Close` should be idempotent.

**Comment by jptosso** [comment]
@jcchavezs could you include test coverage for the close modifications?

**Comment by jcchavezs** [comment]
> @jcchavezs could you include test coverage for the close modifications?

😱 done

**Review by jptosso — APPROVED** [review]

**Pull request merged**